airgun/entities/task.py

```diff
@@ -20,6 +20,22 @@ def read(self, entity_name, widget_names=None):
         view = self.navigate_to(self, 'Details', entity_name=entity_name)
         return view.read(widget_names=widget_names)
 
+    def set_chart_filter(self, char_name, index=None):
+        view = self.navigate_to(self, 'All')
+        if char_name not in view.widget_names:
+            raise ValueError('Provide correct chart name')
+        if char_name == 'StoppedChart' and index:
+            getattr(view, char_name).table[
+                index['row']][index['focus']].click()
+        else:
+            getattr(view, char_name).name.click()
```

Comment:
Common code in both conditional paths is good sign that it can be taken out of conditional:

```python
chart = getattr(view, chart_name)
if index:
    chart.table[index['row']][index['focus']].click()
else:
    chart.name.click()